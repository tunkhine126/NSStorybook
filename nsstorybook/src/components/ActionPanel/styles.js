import { makeStyles } from '@material-ui/core/styles';
import colors from 'global-styles';

export const actionPanelStyles = makeStyles(theme => ({
  cancelBtn: {
    color: colors.TEXT.cancel,
    marginTop: 10,
    '&:hover': {
      color: colors.STATUS.negative,
      backgroundColor: 'transparent',
    },
  },
  panel: {
    background: '#D7D7D7',
    padding: '25px 40px',
    height: '100%',
    [theme.breakpoints.down('md')]: {
      padding: '25px 35px',
    },
  },
  confirmBtn: {
    minHeight: 48,
    width: '100%',
    maxWidth: 225,
    boxShadow: 'none',
    borderRadius: 2,
    fontFamily: 'Barlow',
  },
  progress: {
    color: colors.INTERFACE.ui1,
    padding: 8,
    marginTop: -3,
  },
}));
