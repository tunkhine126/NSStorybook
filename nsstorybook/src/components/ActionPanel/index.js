/**
 *
 * ActionPanel
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import clsx from 'clsx';

import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import CircularProgress from '@material-ui/core/CircularProgress';

import appMessages from 'messages';
import { actionPanelStyles } from './styles';

export function ActionPanel({
  action,
  formId,
  type,
  resetForm,
  disabled,
  loading,
}) {
  const classes = actionPanelStyles();

  return (
    <>
      <Grid item xs={3} xl={2}>
        <div className={clsx(classes.panel)}>
          <Grid
            item
            container
            direction="column"
            justify="center"
            alignItems="center"
          >
            <Button
              variant="contained"
              color="primary"
              size="large"
              className={classes.confirmBtn}
              form={formId}
              type={type}
              disabled={disabled}
            >
              {loading ? (
                <CircularProgress
                  className={classes.progress}
                  color="primary"
                />
              ) : (
                <FormattedMessage {...appMessages[action]} />
              )}
            </Button>
            <Button className={classes.cancelBtn} onClick={resetForm}>
              <FormattedMessage {...appMessages.cancel} />
            </Button>
          </Grid>
        </div>
      </Grid>
    </>
  );
}

export const loadingAction = (message, loading) => {
  const timeOut = setTimeout(() => {
    message(state => ({
      ...state,
      open: false,
      message: '',
    }));
  }, 3000);
  loading(false);
  return timeOut;
};

ActionPanel.propTypes = {
  action: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  formId: PropTypes.string.isRequired,
  resetForm: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  loading: PropTypes.bool,
};

export default ActionPanel;
