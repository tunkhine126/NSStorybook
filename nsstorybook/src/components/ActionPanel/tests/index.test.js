/**
 *
 * Tests for ActionPanel
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import TestingComponent from 'components/shared/Testing';
import ActionPanel from '../Loadable';
import { loadingAction } from '../index';

describe('<ActionPanel />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <ActionPanel
          action="save"
          type="submit"
          formId="test"
          resetForm={() => {}}
          loading
        />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should Execute loadingAction() without errors', () => {
    const spy = jest.spyOn(global.console, 'error');

    const setmessageState = () => {};
    const setloadingState = () => {};

    const timeOut = loadingAction(setmessageState, setloadingState);
    clearTimeout(timeOut);

    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <ActionPanel
          action="save"
          type="submit"
          formId="test"
          resetForm={() => {}}
        />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
