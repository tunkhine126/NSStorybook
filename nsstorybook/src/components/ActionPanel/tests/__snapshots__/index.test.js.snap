// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionPanel /> Should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-xl-2"
>
  <div
    class="makeStyles-panel-164"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-confirmBtn-165 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        form="test"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          <span>
            Save
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-cancelBtn-163"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span>
            Cancel
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
